modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DataNodesManager.java

```diff
@@ -0,0 +1,1527 @@
+                        }
+
+                        int partitionResetDelay = partitionDistributionResetTimeoutSupplier.getAsInt();
+
+                        if (!removedNodes.isEmpty()
+                                && zoneDescriptor.consistencyMode() == HIGH_AVAILABILITY
+                                && partitionResetDelay != INFINITE_TIMER_VALUE) {
+                            reschedulePartitionReset(partitionResetDelay, () -> partitionResetClosure.accept(revision, zoneId), zoneId);
+                        }
+
+                        DistributionZoneTimer mergedScaleUpTimer = mergeTimerOnTopologyChange(zoneDescriptor, timestamp,
+                                scaleUpTimer, addedNodes, newLogicalTopology, true);
+
+                        DistributionZoneTimer mergedScaleDownTimer = mergeTimerOnTopologyChange(zoneDescriptor, timestamp,
+                                scaleDownTimer, removedNodes, newLogicalTopology, false);
+
+                        Pair<HybridTimestamp, Set<NodeWithAttributes>> currentDataNodes = currentDataNodes(
+                                timestamp,
+                                dataNodesHistory,
+                                mergedScaleUpTimer,
+                                mergedScaleDownTimer,
+                                zoneDescriptor
+                        );
+
+                        DistributionZoneTimer scaleUpTimerToSave = timerToSave(timestamp, mergedScaleUpTimer);
+                        DistributionZoneTimer scaleDownTimerToSave = timerToSave(timestamp, mergedScaleDownTimer);
+
+                        boolean addMandatoryEntry = !addedNodesComparingToOldTopology.isEmpty();
+
+                        return new LoggedIif(iif(
+                                and(
+                                        dataNodesHistoryEqualToOrNotExists(zoneId, dataNodesHistory),
+                                        and(
+                                                timerEqualToOrNotExists(zoneScaleUpTimerKey(zoneId), scaleUpTimer),
+                                                timerEqualToOrNotExists(zoneScaleDownTimerKey(zoneId), scaleDownTimer)
+                                        )
+                                ),
+                                ops(
+                                        addNewEntryToDataNodesHistory(zoneId, dataNodesHistory, currentDataNodes.getFirst(),
+                                                currentDataNodes.getSecond(), addMandatoryEntry),
+                                        renewTimer(zoneScaleUpTimerKey(zoneId), scaleUpTimerToSave),
+                                        renewTimer(zoneScaleDownTimerKey(zoneId), scaleDownTimerToSave)
+                                ).yield(true),
+                                ops().yield(false)
+                        ),
+                        updateHistoryLogRecord(zoneId, "topology change", dataNodesHistory, timestamp, currentDataNodes.getFirst(),
```

Comment:
Formatting of this code is all over the place, please make it more clear

Reply:
refactored